### PR TITLE
docs(distribution): New practice to cleanup configuration files

### LINKED ISSUE #8: Cleanup configuration files (if any) when uninstalling the package
This could be a new topic. The CLI application should remove it's internal configuration files (if there are any) when uninstalling the package. We can use NPM's `pre` or `post` uninstall script for this purpose.

https://docs.npmjs.com/misc/scripts

### PR DESCRIPTION
**Pull request by m-sureshraj**
**Changes**

* Added new practice to cleanup configuration files
* Added `.gitignore` file to the project

**Todo**
* I need your help to complete the otherwise section.

Note - Please feel free to correct any grammar mistakes. :)

closes #8

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -315,6 +317,21 @@ References:
 
 </details>
 
+### 2.3 Cleanup configuration files
+
+✅ **Do:**
+Cleanup configuration files when the CLI application gets uninstalled. Optionally CLI application can prompt your users to keep the configuration files to skip the reinitializing phase on the next installation for a better user experience.
```
Comment (lirantal) [inline README.md]:
```suggestion
Cleanup configuration files when the CLI application gets uninstalled. Optionally, CLI applications can prompt your users to keep the configuration files to skip the re-initialising phase on the next installation for a better user experience.
```

Comment (lirantal) [reply README.md]: minor wording suggestions

File: README.md [hunk 2]
```diff
@@ -315,6 +317,21 @@ References:
...
+
+❌ **Otherwise:**
+
+<details>
+  <summary>➡️ <b>Details</b></summary>
+
+As we discussed in the [stateful data section](#13-stateful-data) if your CLI application using storage persistence (configuration files) then the CLI application is responsible for removing its configuration files when it gets uninstalled.
```
Comment (lirantal) [inline README.md]:
```suggestion
As mentioned in the [stateful data section](#13-stateful-data), if your CLI application uses persistent storage , such as to save configuration files, then the CLI application should also be responsible for removing its configuration files when it gets uninstalled.
```

**Review by lirantal — CHANGES_REQUESTED** [review]
Thanks @m-sureshraj 👌

Added some wording suggestions and then we can land this proposal

**Comment by m-sureshraj** [comment]
Hi @lirantal, Thanks for your suggestions. I've updated the pull request, please review it again.

File: README.md [hunk 3]
```diff
@@ -348,6 +350,21 @@ References:
 
 </details>
 
+### 2.3 Cleanup configuration files
+
+✅ **Do:**
+Cleanup configuration files when the CLI application gets uninstalled. Optionally, CLI applications can prompt your users to keep the configuration files to skip the re-initialising phase on the next installation for a better user experience.
+
+❌ **Otherwise:**
```
Comment (m-sureshraj) [inline README.md:356]: Is that okay to keep this section empty?

Comment (m-sureshraj) [inline README.md]: Is that okay to keep this section empty?

Comment (lirantal) [reply README.md:356]: @m-sureshraj we should definitely add this. I didn't notice it was empty before.
Can you please update? I would say that we want to keep the file system clean of any remaining configuration or identifiable data that the CLI tool might have left in when being installed on the system.

**Review by lirantal — CHANGES_REQUESTED** [review]

**Comment by m-sureshraj** [comment]
Hi @lirantal, Sorry for the long... delay.  I've updated the PR, could you please review it again.

File: README.md [hunk 4]
```diff
@@ -335,6 +337,20 @@ References:
 - [Do you really know how a lockfile works for yarn and npm packages?](https://snyk.io/blog/making-sense-of-package-lock-files-in-the-npm-ecosystem/)
 - [Yarn docs: Should lockfiles be committed to the repository?](https://next.yarnpkg.com/advanced/qa#should-lockfiles-be-committed-to-the-repository)
 
+### 2.3 Cleanup configuration files
+
+✅ **Do:**
+Cleanup configuration files when the CLI application gets uninstalled. Optionally, CLI applications can prompt your users to keep the configuration files to skip the re-initialising phase on the next installation for a better user experience.
+
+❌ **Otherwise:**
+We want to keep the file system clean of any remaining configuration or identifiable data that the CLI tool might have left in when being installed on the system.
```
Comment (lirantal) [inline README.md]:
```suggestion
The user's file system may contain residue in the form of orphaned configuration files, and identifiable data which the CLI tool introduced when installed.
```

File: README.md [hunk 5]
```diff
@@ -335,6 +337,20 @@ References:
...
+
+ℹ️ **Details**
+
+As mentioned in the [stateful data section](#13-stateful-data), if your CLI application uses persistent storage, such as to save configuration files, then the CLI application should also be responsible for removing its configuration files when it gets uninstalled.
+
+You can use NPMs `pre` or `post` uninstall [script](https://docs.npmjs.com/misc/scripts) for this purpose. You can find a working example in this [repository](https://github.com/m-sureshraj/jenni/blob/master/src/scripts/pre-uninstall.js).
```
Comment (lirantal) [inline README.md:363]: I like this part, good addition 👌

**Review by lirantal — CHANGES_REQUESTED** [review]
Small wording changes on the `Otherwise` section if you want to commit my suggestion and we're good. Thanks for updating!

File: README.md (hunk 4 above)
Comment (m-sureshraj) [reply README.md]: I like you suggestion.

**Review by lirantal — APPROVED** [review]

**Pull request merged**